### PR TITLE
DT-618: Add in resource and service

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.RestController
+import springfox.documentation.annotations.ApiIgnore
+import uk.gov.justice.digital.hmpps.tokenverification.service.TokenService
+
+@RestController
+@Validated
+@RequestMapping("/token", produces = [MediaType.APPLICATION_JSON_VALUE])
+class TokenResource(private val tokenService: TokenService) {
+  @PostMapping("verify")
+  fun verifyToken(@RequestBody jwt: String): TokenDto = tokenService.verifyToken(jwt)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt]: will add swagger docs later

File: src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  @ApiIgnore
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt]: don't want to expose other services as will be only for auth - need to add spring security to enforce

File: src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  @PostMapping("{authJwtId}")
+  fun addToken(@PathVariable authJwtId: String, @RequestBody jwt: String) {
+    tokenService.addToken(authJwtId, jwt)
+  }
+
+  @ApiIgnore
+  @PostMapping("refresh/{accessJwtId}")
+  fun addRefreshToken(@PathVariable accessJwtId: String, @RequestBody jwt: String) {
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResource.kt]: need separate method for this as can't access the auth jwt id during the refresh process, only the original access token that this refresh token is based off

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
+@file:Suppress("DEPRECATION")
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: What is deprecated in this file?

Comment (petergphillips) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: nothing, was copied from auth.  Will fix in next PR

**Pull request merged**